### PR TITLE
Convert Numeric Values of Environment Variables to Number

### PR DESCRIPTION
**Pull request by mscbpi**
convert numeric values to number

Closes https://github.com/directus/next/issues/363

### CODE REVIEW

**Comment by rijkvanzanten** [comment]
@mscbpi That `isNumeric` function doesn't seem to exist in this file scope.. You might wanna check using `isNaN` or add/import that `isNumeric` function to this file

**Comment by mscbpi** [comment]
> @mscbpi That `isNumeric` function doesn't seem to exist in this file scope.. You might wanna check using `isNaN` or add/import that `isNumeric` function to this file

my bad.

File: api/src/env.ts [hunk 1]
```diff
@@ -62,7 +62,7 @@ function processValues(env: Record<string, any>) {
 		if (value === 'true') env[key] = true;
 		if (value === 'false') env[key] = false;
 		if (value === 'null') env[key] = null;
-		if (isNumeric(value)) env[key] = Number(value);
+		if (!isNaN(Number(value))) env[key] = Number(value);
```
Comment (rijkvanzanten) [inline api/src/env.ts:65]:
```suggestion
		if (isNaN(value) === false) env[key] = Number(value);
```

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]
AFAIK `isNaN` accepts any type for it's parameter, eg `isNaN('12') == false` and `isNaN('rijk') == true`.

Lets update the check to be a little more explicit

**Review by rijkvanzanten — APPROVED** [review]

**Comment by mscbpi** [comment]
With typescript it needs to be typed.

**Pull request merged**